packages/dashboard/src/pages/Wallets/components/AddCollectibleDialog/index.tsx

```diff
@@ -1,48 +1,68 @@
-import { useSnackbarCallback } from '@masknet/shared'
-import { Box, Button, DialogActions, DialogContent, TextField } from '@material-ui/core'
-import { useWallet } from '@masknet/web3-shared'
+import { FormEvent, memo, useEffect, useState } from 'react'
+import { MaskDialog, MaskTextField } from '@masknet/theme'
+import { Box, Button, DialogActions, DialogContent } from '@material-ui/core'
+import {
+    isSameAddress,
+    useERC721ContractDetailed,
+    useERC721TokenDetailedCallback,
+    useWallet,
+} from '@masknet/web3-shared'
 import { EthereumAddress } from 'wallet.ts'
 import { useDashboardI18N } from '../../../../locales'
+import { z } from 'zod'
+import { Controller, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { PluginServices } from '../../../../API'
 
 export interface AddCollectibleDialogProps {
     open: boolean
     onClose: () => void
 }
 
+type FormInputs = {
+    address: string
+    tokenId: string
+}
+
+enum FormErrorType {
+    Added = 'ADDED',
+    NotExist = 'NOT_EXIST',
+}
+
 export const AddCollectibleDialog = memo<AddCollectibleDialogProps>(({ open, onClose }) => {
+    const wallet = useWallet()
     const [address, setAddress] = useState('')
+    const { value: contractDetailed, loading: contractDetailLoading } = useERC721ContractDetailed(address)
+    const [tokenId, setTokenId, erc721TokenDetailedCallback] = useERC721TokenDetailedCallback(contractDetailed)
 
-    const wallet = useWallet()
-    // Todo: The current implement does not work, please refactor this component according to dashboard 1.0 Add Asset:
-    // Todo: maskbook/src/extension/options-page/DashboardDialogs/Wallet/AddERC721Token.tsx
-    // const tokenDetailed = useERC721TokenDetailed(address)
-    // const assetDetailed = useERC721TokenAssetDetailed(tokenDetailed.value)
-
-    const onSubmit = useSnackbarCallback({
-        executor: async () => {
-            // if (!tokenDetailed.value || !assetDetailed.value || !wallet) return
-            // await Promise.all([
-            //     PluginServices.Wallet.addERC721Token({
-            //         ...tokenDetailed.value,
-            //         asset: assetDetailed.value,
-            //     }),
-            //     PluginServices.Wallet.trustERC721Token(wallet.address, tokenDetailed.value),
-            // ])
-        },
-        deps: [wallet],
-        onSuccess: onClose,
-    })
+    const onSubmit = async () => {
+        if (contractDetailLoading || !wallet) return
+
+        const tokenInDB = await PluginServices.Wallet.getERC721Token(address, tokenId)
+        if (tokenInDB) throw new Error(FormErrorType.Added)
+
+        const _tokenDetailed = await erc721TokenDetailedCallback()
+
+        if (
+            (_tokenDetailed && !isSameAddress(_tokenDetailed.info.owner, wallet.address)) ||
+            !_tokenDetailed ||
+            !_tokenDetailed.info.owner
+        ) {
+            throw new Error(FormErrorType.NotExist)
```

Comment:
Is there a UX alarm?

Reply:
The react-hook-form will catch the error and I will show the error message on UI.
```
handleSubmit(onSubmit)(event).catch((error) => {
                setError('tokenId', {
                    type: 'value',
                    message:
                        error.message === FormErrorType.Added
                            ? t.wallets_collectible_been_added()
                            : t.wallets_collectible_error_not_exist(),
                })
            })
```